compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -223,26 +223,51 @@ static LogicalResult setRootConfigForPackPipeline(func::FuncOp entryPointFn,
   return success();
 }
 
+/// Utility function that, given an element type, computes tiling scaling factor
+/// as : 64/bitWidth.
+/// Therefore, for example, if the element type has :-
+///     a. 64 bit width -> scaling factor will be 1.
+///     b. 32 bit width -> scaling factor will be 2.
+///     c. 16 bit width -> scaling factor will be 4.
+/// This can easily be scaled to other element data types by just changing the
```

Comment:
Don't you also need to set scaling factor according to data types, i.e., int or float?

Reply:
Nope. It should only be based on the memory footprint which in turn is based on the bitwidth.

Eg: `fp16` uses 16 bit to represent a data. `int16` uses 16 bit to represent a data. Therefore their memory footprint is the same.